django/db/models/lookups.py

```diff
@@ -362,7 +362,7 @@ def process_rhs(self, compiler, connection):
 class IsNull(BuiltinLookup):
     lookup_name = 'isnull'
 
-    def as_sql(self, compiler, connection):
+    def as_sql(self, compiler, connection, **kwargs):
         sql, params = compiler.compile(self.lhs)
```

Comment:
`template` `function` and `arg_joiner` are only common to `Func` expressions. But `Func` expressions also allow arbitrary kwargs as input into `template` interpolation. Making this change in the general case is not right. I'll comment specifically on other entries.